components/permalink_redirector/actions.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getChannel} from 'mattermost-redux/actions/channels';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getTeam as getTeamSelector} from 'mattermost-redux/selectors/entities/teams';
+import {getPost} from 'mattermost-redux/actions/posts';
+import {getTeam} from 'mattermost-redux/actions/teams';
+
+import {redirectUserToDefaultTeam} from 'actions/global_actions.jsx';
+
+import {Constants, ErrorPageTypes} from 'utils/constants.jsx';
+import {browserHistory} from 'utils/browser_history';
+
+import LocalStorageStore from 'stores/local_storage_store';
+
+export const redirect = (postId) => {
+    return async (innerDispatch, getState) => {
+        const state = getState();
+        let team = null;
+
+        const post = await innerDispatch(getPost(postId));
```

Comment:
I really like that this works for any post, not just direct messages, but I am worried that this logic could easily regress given the need to duplicate some of these checks elsewhere in the application. 0/5, but I'd be fine with simplifying this to only work for DM posts since they have, by definition, no associated team.

Reply:
Updated to only support DM permalinks, or any other team agonistic urls.